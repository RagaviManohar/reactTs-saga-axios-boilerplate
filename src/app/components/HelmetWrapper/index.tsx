import { Helmet } from 'react-helmet-async';

interface HelmetWrapperProps {
  title: string;
  description?: string;
}

export const HelmetWrapper = ({ title, description }: HelmetWrapperProps) => (
  <Helmet>
    <title>{title}</title>
    {description && <meta name="description" content={description} />}
  </Helmet>
);

export default HelmetWrapper;
