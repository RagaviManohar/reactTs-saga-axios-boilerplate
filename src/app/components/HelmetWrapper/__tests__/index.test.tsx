import { render, cleanup, MockHelmetWrapper } from 'jest.setup';

import { HelmetWrapper } from 'app/components/HelmetWrapper/index';

afterEach(cleanup);

const renderComponent = () =>
  render(
    <MockHelmetWrapper>
      <HelmetWrapper title="home" description="home page" />
    </MockHelmetWrapper>,
  );

describe('<HelmetWrapper  />', () => {
  it('expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderComponent();
    expect(spy).not.toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
