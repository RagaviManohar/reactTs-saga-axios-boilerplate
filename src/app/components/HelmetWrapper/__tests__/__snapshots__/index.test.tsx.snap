// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelmetWrapper  /> should match snapshot 1`] = `null`;
